mypyc/irbuild/for_helpers.py

```diff
@@ -131,7 +132,41 @@ def for_loop_helper_with_index(builder: IRBuilder, index: Lvalue, expr: Expressi
     builder.activate_block(exit_block)
 
 
+def preallocate_space_helper(builder: IRBuilder,
+                             gen: GeneratorExpr,
+                             empty_op_llbuilder: Callable[[Value, int], Value],
+                             set_item_op: CFunctionDescription) -> Optional[Value]:
+    """Currently we only optimize for simplest generator expression.
+
```

Comment:
Add an example of what sort things this can be used to handle (e.g. `tuple(x + 1 for x in a)` if `a` is a list).